libs/api/domains/file-upload/src/lib/file-upload.resolver.ts

```diff
@@ -1,12 +1,47 @@
-import { Args, Resolver, Mutation } from '@nestjs/graphql'
+import { Args, Resolver, Mutation, Query } from '@nestjs/graphql'
 import { PresignedPost } from './presignedPost.model'
 import { FileStorageService } from '@island.is/file-storage'
 import { PresignedPost as S3PresignedPost } from '@aws-sdk/s3-presigned-post'
+import { MalwareScanStatus } from './malwareScanStatus.model'
+
+const MALWARE_FETCH_ATTEMPTS = 4
+const MALWARE_MIN_MS_BETWEEN_FETCHES = 1000
 
 @Resolver()
 export class FileUploadResolver {
   constructor(private fileStorageService: FileStorageService) {}
 
+  @Query(() => MalwareScanStatus)
+  async malwareScanStatus(
+    @Args('filename') filename: string,
+  ): Promise<MalwareScanStatus> {
+    let scanStatus = undefined
+    for (let i = 0; i < MALWARE_FETCH_ATTEMPTS; i++) {
+      const waitTimeThisLoop = MALWARE_MIN_MS_BETWEEN_FETCHES * (Math.pow(2, i))
+      await new Promise((resolve) => setTimeout(resolve, waitTimeThisLoop))
```

Comment:
~Waiting immediately assumes that this resolver is always called right after upload finished. Good to check tags once before doing backoff retries for better performance when the scan is already done.